### PR TITLE
feat: allow user to use github app to authenticate

### LINKED ISSUE #89: [Feature Request] Allow GitHub App Installation Token instead of user PAT Token
To ensure the GitHub action is not dependent on a single user's PAT Token, I'd like to add functionality to allow for a GitHub App Installation Token.  ~This may require a switch from github3.py though.  I wasn't seeing it as an option with github3.py (still looking).~ github3.py handles app authentication, will stick with the same lib 😄 

> a GitHub App Installation Token is an efficient way to unlock the maximum alotted API calls to avoid rate limiting and can be used by multiple members of your organization

from [Community Discussions](https://github.com/orgs/community/discussions/48186#:~:text=a%20GitHub%20App%20Installation%20Token%20is%20an%20efficient%20way%20to%20unlock%20the%20maximum%20alotted%20API%20calls%20to%20avoid%20rate%20limiting%20and%20can%20be%20used%20by%20multiple%20members%20of%20your%20organization)

~[PyGitHub](https://github.com/PyGithub/PyGithub) is what I've used in the past.~

[GitHub App Installation Authentication](https://pygithub.readthedocs.io/en/stable/examples/Authentication.html#app-installation-authentication)

### PR DESCRIPTION
**Pull request by jmeridth**
Fixes #89 

# Pull Request

## Proposed Changes

This feature allows users to authenticate with either just a GitHub app or a GitHub App Installation.

- [x] setup authentication
- [x] helper function to get integer environment variables without ValueError
- [x] tests
  - [x] GitHub App Installation passing
  - [x] helper function to get integer environment variables
- [x] added .venv to flake8 exlusion list (may need to address this another way)
- [x] alphabetized lists in places (README, .env-sample)

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer

- [x] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, or `breaking`

### CODE REVIEW

**Comment by jmeridth** [comment]
@zkoppert working through one more set of tests but welcoming feedback early.

- could probably mock out the test pem/rsa better (one in there is random generated)
- if you know of a Pythonic one-liner for getting integer environment variable that can handle `int('')` which results in `ValueError`, I welcome the feedback.

File: test_stale_repos.py [hunk 1]
```diff
@@ -58,8 +61,54 @@ class AuthToGithubTestCase(unittest.TestCase):
 
     """
 
+    TEST_GH_APP_PRIVATE_KEY = """-----BEGIN RSA PRIVATE KEY-----
```
Comment (jmeridth) [inline test_stale_repos.py:64]: This is fake.

Comment (jmeridth) [reply test_stale_repos.py:64]: Got flagged by gitguardian. Definitely gonna go mock route in this. Lol

Comment (jmeridth) [reply test_stale_repos.py]: Removed

**Comment by jmeridth** [comment]
@zkoppert Curious your thoughts on this one.  Could we please give a tests a run.  Curious if any diff upstream compared to my local runs.  Thank you.

**Comment by jmeridth** [comment]
Looking at lint errors.

**Comment by jmeridth** [comment]
More linting fixes.  In another PR I'm going to get superlinter working with `make lint`.  My `make lint` passes locally but the SuperLinter GitHub Action is finding more stuff.

File: README.md [hunk 2]
```diff
@@ -171,6 +174,36 @@ jobs:
           INACTIVE_DAYS: 365
 ```
 
+### Authenticating with a GitHub App and Installation
+
+You can authenticate as a GitHub App Installation by providing additional environment variables.
```
Comment (zkoppert) [inline README.md:179]:
```suggestion
You can authenticate as a GitHub App Installation by providing additional environment variables. If `GH_TOKEN` is set alongside these GitHub App Installation variables, the `GH_TOKEN` will be ignored and not used.
```

**Review by zkoppert — APPROVED** [review]

**Comment by zkoppert** [comment]
Great work! Thanks for adding this! It's got me thinking about all the other actions that could use this 😄

**Comment by jmeridth** [comment]
@zkoppert I was thinking the same.  The `codeowners` one is on my list.  Want me to do a PR there?

**Comment by jmeridth** [comment]
@zkoppert Was thinking about keeping it DRY somehow also. 🤔

**Comment by zkoppert** [comment]
> @zkoppert I was thinking the same.  The [`codeowners`](https://github.com/github/cleanowners) one is on my list.  Want me to do a PR there?

That would be great!

**Pull request merged**